### PR TITLE
[FEATURE] Enable large tensor support for insert

### PR DESCRIPTION
**Pull request by access2rohit**
## Description ##
Enable LTS for numpy insert op for 2/3 insert kernels(`insert_scalar` and `insert_slice`). Performance of `insert_tensor` needs to be optimized first before enabling large tensor for it. Currently if the user attempts to use `insert_tensor` the program will give error message that large tensor is not supported for insert_tensor kernel.

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Testing ###
```
(pytest) ubuntu@ip-172-31-90-243 ~/workspace/incubator-mxnet (insert_lt) $ python -m pytest -s --exitfirst --verbose tests/nightly/test_np_large_array.py::test_insert
==================================================================================================================================== test session starts ====================================================================================================================================
platform linux -- Python 3.6.10, pytest-5.3.5, py-1.8.2, pluggy-0.13.1 -- /home/ubuntu/anaconda3/envs/pytest/bin/python
cachedir: .pytest_cache
rootdir: /home/ubuntu/workspace/incubator-mxnet, inifile: pytest.ini
plugins: timeout-1.4.2
timeout: 1200.0s
timeout method: signal
timeout func_only: False
collected 1 item

tests/nightly/test_np_large_array.py::test_insert Setting module np/mx/python random seeds, use MXNET_MODULE_SEED=1610131494 to reproduce.
[20:31:33] ../src/storage/storage.cc:199: Using Pooled (Naive) StorageManager for CPU
PASSED

===================================================================================================================================== warnings summary ======================================================================================================================================
tests/nightly/test_np_large_array.py:92
  /home/ubuntu/workspace/incubator-mxnet/tests/nightly/test_np_large_array.py:92: DeprecationWarning: invalid escape sequence \
    '''

tests/nightly/test_np_large_array.py:1243
  /home/ubuntu/workspace/incubator-mxnet/tests/nightly/test_np_large_array.py:1243: DeprecationWarning: invalid escape sequence \
    '''

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================================================================================================================= 1 passed, 2 warnings in 68.90s (0:01:08) ==========================================================================================================================
```

### CODE REVIEW

File: src/operator/numpy/np_insert_op_tensor-inl.h [hunk 1]
```diff
@@ -175,34 +175,34 @@ void NumpyInsertTensorCompute(const nnvm::NodeAttrs& attrs,
...
     Tensor<xpu, 1, char> temp_mem =
       ctx.requested[0].get_space_typed<xpu, 1, char>(Shape1(temp_mem_size), s);
     int64_t* indices_ptr = reinterpret_cast<int64_t*>(temp_mem.dptr_);
     int64_t* sorted_indices_ptr = reinterpret_cast<int64_t*>(indices_ptr + indices_len);
-    int* order_ptr = reinterpret_cast<int*>(sorted_indices_ptr + indices_len);
-    int* is_insert = reinterpret_cast<int*>(order_ptr + indices_len);
-    int* origin_idx = reinterpret_cast<int*>(is_insert + outshape[axis]);
+    index_t* order_ptr = reinterpret_cast<index_t*>(sorted_indices_ptr + indices_len);
+    index_t* is_insert = reinterpret_cast<index_t*>(order_ptr + indices_len);
+    index_t* origin_idx = reinterpret_cast<index_t*>(is_insert + outshape[axis]);
```
Comment (Zha0q1) [inline src/operator/numpy/np_insert_op_tensor-inl.h:196]: would you explain a bit why this pointer conversion would work? `sorted_indices_ptr` is defined as `int64_t*` and if the pointers here are int* then the stride is different?

File: src/operator/numpy/np_insert_op_slice-inl.h [hunk 2]
```diff
@@ -147,10 +147,11 @@ void NumpyInsertSliceCompute(const nnvm::NodeAttrs& attrs,
   } else {
     // broadcast check
     for (int i = outshape.ndim() - 1; i >= 0; --i) {
-      int sz = outshape[i];
+      index_t sz = outshape[i];
       if (i == axis) {
         sz = numnew;
       }
+      LOG(INFO) << "values.shape_[" << i << "]=" << values.shape_[i] << ", sz= " << sz;
```
Comment (Zha0q1) [inline src/operator/numpy/np_insert_op_slice-inl.h:154]: we can probably remove this now?

**Review by Zha0q1 — COMMENTED** [review]
left two questions, rest looks good! Thanks for the fix, this is a hard one

File: src/operator/numpy/np_insert_op_tensor-inl.h (hunk 1 above)
Comment (access2rohit) [reply src/operator/numpy/np_insert_op_tensor-inl.h:196]: this is how you allocate temporary workspace(with different data types). order_ptr was `int*` before now i have changed it to `index_t*` and updated memory to be allocated for the order_ptr as well to `index_t`. Stride is for pointer index and not for data.

Comment (Zha0q1) [reply src/operator/numpy/np_insert_op_tensor-inl.h:196]: I see, we are using different sections of that temporary mem

**Comment by Zha0q1** [comment]
@access2rohit are you going to include `insert_tensor` and `delete` in this pr too?

File: src/operator/numpy/np_insert_op_tensor-inl.h (hunk 1 above)
Comment (access2rohit) [reply src/operator/numpy/np_insert_op_tensor-inl.h:196]: correct

**Comment by access2rohit** [comment]
@Zha0q1 i had to fix funxtions in delete since they were used by insert. I have fixed insert_tensor as well but i needs to be optimized to be even able to test it properly. For input sizes over 10million it takes forever even w/ int32 indexing.

**Comment by access2rohit** [comment]
@mxnet-label-bot update [pr-awaiting-review]

File: tests/python/unittest/test_numpy_op.py [hunk 3]
```diff
@@ -3697,7 +3697,7 @@ def GetNdim(tp):
             np_out = _np.insert(a.asnumpy(), obj_onp, b.asnumpy(), axis=axis)
 
             assert_almost_equal(mx_out.asnumpy(), np_out, rtol=1e-3, atol=1e-5)
-
+test_np_insert()
```
Comment (Zha0q1) [inline tests/python/unittest/test_numpy_op.py:3700]: Why do we need to call it manually?

**Comment by Zha0q1** [comment]
Left 1 comment, rest looks good!

**Review by Zha0q1 — APPROVED** [review]

File: tests/nightly/test_np_large_array.py [hunk 4]
```diff
@@ -2035,6 +2035,7 @@ def test_vstack():
 
 
 @use_np
+<<<<<<< HEAD
```
Comment (sandeep-krishnamurthy) [inline tests/nightly/test_np_large_array.py:2038]: merge conflicts.

Comment (access2rohit) [reply tests/nightly/test_np_large_array.py:2038]: aahh .. missed it !

**Review by sandeep-krishnamurthy — APPROVED** [review]

**Pull request merged**